pkg/booter/bootentry.go

```diff
@@ -0,0 +1,74 @@
+package booter
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/insomniacslk/systemboot/pkg/vpd"
+)
+
+// BootEntry represents a boot entry, with its name, configuration, and Booter
+// instance. It can map to existing key-value stores like VPD or EFI vars.
+type BootEntry struct {
+	Name   string
+	Config []byte
+	Booter Booter
+}
+
+var supportedBooterParsers = []func([]byte) (Booter, error){
+	NewNetBooter,
+}
+
+// GetBooterFor looks for a supported Booter implementation and returns it, if
+// found. If not found, a NullBooter is returned.
+func GetBooterFor(entry BootEntry) Booter {
+	var (
+		booter Booter
+		err    error
+	)
+	for idx, booterParser := range supportedBooterParsers {
+		log.Printf("Trying booter #%d", idx)
+		booter, err = booterParser(entry.Config)
+		if err != nil {
+			log.Printf("This config is not valid for this booter (#%d)", idx)
+			continue
+		}
+		break
+	}
+	if booter == nil {
+		log.Printf("No booter found for entry: %+v", entry)
+		return &NullBooter{}
+	}
+	return booter
+}
+
+// GetBootEntries returns a list of BootEntry objecsts stored in the VPD
+// partition of the flash chip
+func GetBootEntries() []BootEntry {
+	var bootEntries []BootEntry
+	for idx := 0; idx < 9999; idx++ {
+		key := fmt.Sprintf("Boot%04d", idx)
+		// try the RW entries first
+		value, err := vpd.Get(key, false)
+		if err == nil {
+			bootEntries = append(bootEntries, BootEntry{Name: key, Config: value})
```

Comment:
They are not skipped, they are checked first. If no RW variable with that name is found it will try a RO variable. Basically I'm giving priority to RW variables so that RO ones are used as fallback